src/collapse/vsprof.rs

```diff
@@ -0,0 +1,188 @@
+                //    until the top node is our parent
+                match prev_depth.cmp(&depth) {
+                    // Case 1
+                    Ordering::Less => {
+                        self.stack
+                            .push((function_name.to_string(), number_of_calls));
+                    }
+                    // Case 2
+                    Ordering::Equal => {
+                        self.write_stack(occurences);
+                        self.stack.pop();
+                        self.stack
+                            .push((function_name.to_string(), number_of_calls));
+                    }
+                    // Case 3
+                    Ordering::Greater => {
+                        self.write_stack(occurences);
+
+                        // Pop (prev_depth - depth - 1) nodes of the stack, this leaves our
+                        // parent as the top node.
+                        for _ in 0..=prev_depth - depth {
+                            self.stack.pop();
+                        }
+
+                        self.stack
+                            .push((function_name.to_string(), number_of_calls));
+                    }
+                }
+            }
+            _ => {
+                return invalid_data_error!("Unable to parse function name from line:\n{}", line);
+            }
+        }
+
+        Ok(())
+    }
+
+    // Store the current stack in `occurences`
+    fn write_stack(&self, occurences: &mut Occurrences) {
+        let (functions, samples): (Vec<_>, Vec<_>) = self.stack.iter().cloned().unzip();
+        if let Some(number_of_calls) = samples.last() {
+            if *number_of_calls > 0 {
+                occurences.insert(functions.join(";"), *number_of_calls);
+            }
+        }
+    }
+}
+
+/// Gets the number from the start of the line. This can either be a number <1000, in which case the
+/// line doesn't contain double quotes, or the number can be >1000, in which case the line does
+/// contain double quotes. In both cases `line` may start with a leading comma, which will be
+/// ignored.
+fn get_next_number(line: &str) -> io::Result<(usize, &str)> {
+    // Trim the leading comma, if any
+    let line = line.trim_start_matches(',');
+
+    if let Some('"') = line.chars().next() {
+        // number >1000
+        // Trim the leading double quote and then split on the next double quote, which leaves
+        // the number as the first part of the result.
+        let mut split = line.trim_start_matches('"').splitn(2, '"');
+        if let Some(num) = split.next() {
+            // If `num` contains both a comma and a period, it is definitely a fractional
+            // number, regardless of the current culture. As a fractional number of calls is
+            // not possible, we don't accept them and fallback to the error returned below.
+            if !(num.contains('.') && num.contains(',')) {
```

Comment:
This condition confuses me. A number that contains _neither_ is accepted here, as is a number like `"100.53"`, which doesn't seem right? Is `,` _always_ used as the thousands separator in this file format, or only sometimes?

Reply:
What I was trying to do here (in maybe not the best way possible), was to disallow fractional numbers, but allow numbers with thousands separators, regardless of the current culture. I have looked, but I'm unable to find whether VS always outputs with `,`s as thousands separators, or whether they'll use `.`s if that is required by the current culture. They do seem to always use thousands separators if the number is >1000.

Right now I'm only removing the `,`s, because apparently Rust doesn't support them during parsing (probably has to do with culture stuff). Do you know of a more idiomatic way of handling numbers which contain thousands separators, regardless of culture? Or are you fine with just assuming that VS always uses `,`s?